module/Activity/src/Activity/Service/Activity.php

```diff
@@ -110,12 +111,30 @@ public function createActivity(array $params)
         // Find the creator
         $user = $this->getServiceManager()->get('user_role');
         if ($user === 'guest') {
-            throw new \InvalidArgumentException('Guests can not create activities');
+            throw new NotAllowedException('Guests can not create activities');
         }
-        $params['creator'] = $user;
-
         $activity = new ActivityModel();
-        $activity->create($params);
+        $activity->setBeginTime(new \DateTime($params['beginTime']));
+        $activity->setEndTime(new \DateTime($params['endTime']));
+        $activity->setName($params['name']);
+        $activity->setNameEn($params['name_en']);
+        $activity->setLocation($params['location']);
+        $activity->setLocationEn($params['location_en']);
+
+        if (!$params['costs_unknown']) {
+            $activity->setCosts($params['costs']);
+            $activity->setCostsEn($params['costs_en']);
+        }
+
+        $activity->setDescription($params['description']);
+        $activity->setDescriptionEn($params['description_en']);
+        $activity->setCanSignUp($params['canSignUp']);
+
+        // Not user provided input
+        $activity->setCreator($user);
```

Comment:
I just looked at this and I figured out the actual causes of the issue:
1. Your ORM annotations aren't correct, nullable should be listed inside the JoinColumn statement (don't ask me why)
2. The authentication service returns a User\Model\User object, however this object is not known to doctrine and therefore doctrine can't do anything with it. $em->merge(..) fixes that.

The following patch fixes the issue: https://gist.github.com/jszanto/97741da52d9407e2e5fe